requirements.txt

```diff
@@ -0,0 +1,2 @@
+pre-commit
```

Comment:
Since we already have a pyproject.toml we don't need to add a requirements.txt file, also poetry should not be installed via pip, but rather via `pipx`globally and not for for the project.

Suggestion: remove this file, run `poetry add --group dev pre-commit` and do `poetry run pre-commit install` after the `poetry install` command in the makefile?

Reply:
Just wanted to jump in and say it took me like an hour to have the setup complete (pyenv failed constantly saying I was missing libraries but actually it was fine) to be able to run make setup and then wonder why it failed. Also I don't like installing anything globally and did everything in a local venv. I normally just run virtualenv and use requirements.txt and fix the library versions in ranges. Also as far as I've read it would be even more of a pain doing the setup in Windows.

Reply:
seting up in ubuntu 22 took:
- running `curl https://pyenv.run | bash`
- finding the files to setup `Then, if you have ~/.profile, ~/.bash_profile or ~/.bash_login, add the commands there as well. If you have none of these, add them to ~/.profile.`
- appending to that file `echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
echo 'eval "$(pyenv init -)"' >> ~/.profile` and the other configs specified after running the first curl step
- trying to run `pyenv install $(cat .python-version)` and it failing multiple times.
  - finding what libraries were missing and which not.
  - finding remote apt repositories to install from
  - installing the -dev versions of each
- scratch my head at why make setup wasn't working
- setup virtualenv and activate pyenv python 3.11.9 version
- pip install pre-commit poetry

If it was this complicated on a unix system...
The windows pyenv even has its [own github project](https://github.com/pyenv-win/pyenv-win) 

A docker image to run the tests in might make this all easier. 
Another option is to have something that generates a requirements.txt file from the poetry.lock.  And since most ppl. will have one of the 3 latest Python versions on their system most recent version of libraries support several python versions. And also the fact that the current project dependencies are only Numpy and PyArrow. Adding the fact that I don't think we'll have that many more dependencies since we will most likely be bootstrapping everything here.

Just wanted to clarify that I do appreciate the amount of effort that goes into structuring a project properly like you did. I just wanted to share my feedback in things that can be made more user friendly.

Reply:
you are assuming pipx and poetry are preinstalled?
is the plan to install everything from scratch including python, mojo and all min set of  python packages into python virtual environment managed by poetry?
what are the target systems? ubuntu, macos and windows?